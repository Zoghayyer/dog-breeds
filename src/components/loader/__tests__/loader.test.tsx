import React from 'react';
import { shallow } from 'enzyme';
import Loader, { LoaderProps } from '../loader';

describe('(Component) Loader', () => {
  let component;
  const defaultProps: LoaderProps = {
    className: 'className',
    size: 'sm',
  };

  const shallowLoader = (extendedProps: Partial<LoaderProps> = {}) => {
    component = shallow(<Loader {...defaultProps} {...extendedProps} />);
  };

  beforeEach(() => shallowLoader());

  it('renders without crashing', () => {
    shallow(<Loader {...defaultProps} />);
    expect(component).toMatchSnapshot();
  });
});
