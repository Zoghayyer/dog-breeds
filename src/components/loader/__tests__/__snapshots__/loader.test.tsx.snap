// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`(Component) Loader renders without crashing 1`] = `
<div
  className="row d-flex justify-content-center className"
>
  <Spinner
    animation="border"
    className="loader-color"
    role="status"
    size="sm"
  >
    <span
      className="sr-only"
    >
      Loading...
    </span>
  </Spinner>
</div>
`;
